### PR TITLE
Add submissionId to logs where it has been missing [DPP-1052]

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala [hunk 1]
```diff
@@ -92,47 +92,51 @@ private[apiserver] final class ApiPartyManagementService private (
...
                 )
-              }
-          })
-          .andThen(logger.logErrorsOnCall[AllocatePartyResponse])
+              )
+            }
+        })
+        .andThen(logger.logErrorsOnCall[AllocatePartyResponse])
     }
+  }
```
Comment (skisel-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:139]: my attempt to make less of nesting
```
  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] = {
    val submissionId = submissionIdGenerator(request.partyIdHint)
    withEnrichedLoggingContext(
      logging.partyString(request.partyIdHint),
      logging.submissionId(submissionId),
    ) { implicit loggingContext =>
      logger.info("Allocating party")
      implicit val telemetryContext: TelemetryContext =
        DefaultTelemetry.contextFromGrpcThreadLocalContext()

      val submit = for {
        partyIdHint <- validatedPartyIdHint(request)
        displayName = Some(request.displayName).filterNot(_.isEmpty)
        input = (partyIdHint, displayName)
        PartyEntry.AllocationAccepted(_, partyDetails) <- synchronousResponse.submitAndWait(
          submissionId,
          input,
        )
      } yield {
        val details = PartyDetails(
          partyDetails.party,
          partyDetails.displayName.getOrElse(""),
          partyDetails.isLocal,
        )
        AllocatePartyResponse(Some(details))
      }
      submit.andThen(logger.logErrorsOnCall[AllocatePartyResponse])
    }
  }

  private def validatedPartyIdHint(request: AllocatePartyRequest): Future[Option[IdString.Party]] =
    if (request.partyIdHint.isEmpty)
      Future.successful(None)
    else
      Ref.Party
        .fromString(request.partyIdHint)
        .fold(
          toError(request),
          party => Future.successful(Some(party)),
        )

  private def toError(request: AllocatePartyRequest)(error: String) =
    Future.failed(
      ValidationLogger.logFailure(request, ValidationErrors.invalidArgument(error))
    )
```

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala [hunk 2]
```diff
@@ -92,47 +92,51 @@ private[apiserver] final class ApiPartyManagementService private (
...
-              .map { case PartyEntry.AllocationAccepted(_, partyDetails) =>
-                AllocatePartyResponse(
-                  Some(
-                    PartyDetails(
-                      partyDetails.party,
-                      partyDetails.displayName.getOrElse(""),
-                      partyDetails.isLocal,
-                    )
+  override def allocateParty(request: AllocatePartyRequest): Future[AllocatePartyResponse] = {
+    val submissionId = submissionIdGenerator(request.partyIdHint)
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:96]: Can you also move the `contextualizedErrorLogger` in scope and pass this newly generated submissionId to it?

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:96]: That one is used in the `ValidationLogger.logFailure` below on line 114.

**Review by tudor-da — APPROVED** [review]
Left one comment. Otherwise lgtm provided that the failing conformance test is fixed.

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala (hunk 2 above)
Comment (mziolekda) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPartyManagementService.scala:96]: ok

**Pull request merged**